org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java

```diff
@@ -91,4 +112,76 @@ private static org.eclipse.jface.text.Position toPosition(Position position, IDo
 		return new org.eclipse.jface.text.Position(start, 1);
 	}
 
+	@Override
+	public final Consumer<MouseEvent> getAction() {
+		return me -> {
+			String title= getLabel();
+			if (title != null && !title.isEmpty()) {
+				findLabelPart(me).ifPresent(labelPart -> {
+					final Command command = labelPart.getCommand();
+					if(command != null && command.getCommand() != null && !command.getCommand().isEmpty()) {
+						ExecuteCommandOptions provider = wrapper.getServerCapabilities().getExecuteCommandProvider();
+						if (provider != null && provider.getCommands().contains(command.getCommand())) {
+							wrapper.execute(ls -> ls.getWorkspaceService()
```

Comment:
should we not use LanguageServerDocumentExecutor.LanguageServerDocumentExecutor(IDocument) and execute the command through it so that we ensure the order of the executions for the given document?

Reply:
Shall I change it to:

```java
	public final Consumer<MouseEvent> getAction() {
		return me -> {
			String title= getLabel();
			if (title != null && !title.isEmpty()) {
				findLabelPart(me).ifPresent(labelPart -> {
					final Command command = labelPart.getCommand();
					if(command != null && command.getCommand() != null && !command.getCommand().isEmpty()) {
						ExecuteCommandOptions provider = wrapper.getServerCapabilities().getExecuteCommandProvider();
						if (provider != null && provider.getCommands().contains(command.getCommand())) {
							LanguageServerDocumentExecutor.forDocument(document).computeAll((wrapper, ls) -> {
								if (wrapper == this.wrapper) {
									return ls.getWorkspaceService()
											.executeCommand(new ExecuteCommandParams(command.getCommand(), command.getArguments()));
								}
								return CompletableFuture.completedFuture(null);
							});
						} else  {
							CommandExecutor.executeCommandClientSide(command, document);
						}
					}
				});
			}
		};
	}

```

Reply:
Sorry for having missed your comment. Yes, I think something like that is good. Maybe use just `(w, ls)` instead of `(wrapper, ls)`, as I think that is the commonly used abbreviation in the codebase.

Reply:
IMO, explicit names that one can read out of a context are always better than abbreviation which require deep knowledge of the overall code. I think we should avoid mandating such "shortening" of names in general and most often stick with the contributor's style. I find project that welcome and handle more diverse style are more comfortable to work with than the ones that force a style (Maven is my worst experience in OSS contribution because of that)

Reply:
Yes, I agree that they are usually better. Only this context, `wrapper` already exists as a class field, so I would tend to give a different name if possible. In any case, it was only a suggestion, not a requirement :)

Reply:
>  Only this context, wrapper already exists as a class field,

Ah OK, I had missed that point.